substrate/frame/support/src/lib.rs

```diff
@@ -936,6 +936,83 @@ pub mod pallet_prelude {
 	pub use sp_weights::Weight;
 }
 
+/// The pallet macro has 2 purposes:
+///
+/// * [For declaring a pallet as a rust module](#1---pallet-module-declaration)
+/// * [For declaring the `struct` placeholder of a
+///   pallet](#2---pallet-struct-placeholder-declaration)
+///
+/// # 1 - Pallet module declaration
+///
+/// The module to declare a pallet is organized as follow:
+/// ```
+/// #[frame_support::pallet]    // <- the macro
+/// mod pallet {
+/// 	#[pallet::pallet]
+/// 	pub struct Pallet<T>(_);
+///
+/// 	#[pallet::config]
+/// 	pub trait Config: frame_system::Config {}
+///
+/// 	#[pallet::call]
+/// 	impl<T: Config> Pallet<T> {
+/// 	}
+///
+/// 	/* ... */
+/// }
+/// ```
+///
+/// The documentation for each individual part can be found at [frame_support::pallet_macros]
```

Comment:
Does this result in a link?

Reply:
Yes, I think rust doc link directly to item when there are in scope.